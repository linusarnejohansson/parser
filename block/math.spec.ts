import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.math", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("$$\na^2 + b^2 = c^2$$\nMath Caption."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toEqual([
			{
				class: "Block.Math",
				value: "a^2 + b^2 = c^2",
				content: [
					{
						class: "Inline.Text",
						value: "Math Caption.",
					},
				],
			},
		])
	})
})
