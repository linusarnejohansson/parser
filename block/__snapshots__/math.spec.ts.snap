// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.math simple 1`] = `
Array [
  Object {
    "class": "Block.Math",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Math Caption.",
      },
    ],
    "value": "a^2 + b^2 = c^2",
  },
]
`;
